source/codegen/templates/service.cpp.mako

```diff
@@ -13,6 +13,7 @@ custom_types = common_helpers.get_custom_types(config)
 has_async_functions = any(service_helpers.get_async_functions(functions))
 has_two_dimension_functions = any(service_helpers.get_functions_with_two_dimension_param(functions))
 function_names = service_helpers.filter_proto_rpc_functions_to_generate(functions)
+streaming_functions = common_helpers.filter_streaming_functions(functions)
```

Comment:
Rename this to `streaming_functions_to_generate`. Then pass `functions_to_generate` and `functions` to `filter_streaming_functions` API so that it returns only those functions which need to be generated.

Places where you check `% if streaming_functions` below will return true even if all functions in `streaming_functions` are marked false for generation and you will end up generating code when there are no streaming functions generated in cpp file.

Reply:
Correct. Updated